nominal/core/asset.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from types import MappingProxyType
+from typing import Mapping, Protocol, Sequence
+
+from typing_extensions import Self
+
+from nominal._api.combined import scout_asset_api, scout_assets
+from nominal.core._clientsbunch import HasAuthHeader
+from nominal.core._utils import HasRid, update_dataclass
+from nominal.ts import IntegralNanosecondsUTC
+
+
+@dataclass(frozen=True)
+class Asset(HasRid):
+    rid: str
+    name: str
+    description: str | None
+    properties: Mapping[str, str]
+    labels: Sequence[str]
+
+    _clients: _Clients = field(repr=False)
+
+    class _Clients(HasAuthHeader, Protocol):
+        @property
+        def assets(self) -> scout_assets.AssetService: ...
+
+    def update(
+        self,
+        *,
+        name: str | None = None,
```

Comment:
Should this mirror the underlying API and be called `title`?

Reply:
We use name and title interchangeably at the API layer, but made the decision to unify this into "name" in the client lib regardless of what the API is using.